tests/unit/test-util/test.artifacts.js

```diff
@@ -55,4 +56,43 @@ describe('prepareArtifactsDir', () => {
     }
   ));
 
+  it('throws error when checking the directory if lacks writing permissions',
+    () => withTempDir(
+    (tmpDir) => {
+      const tmpPath = path.join(tmpDir.path(), 'build');
+      return fs.mkdir(tmpPath, '0622').then(() => {
+        const artifactsDir = path.join(tmpPath, 'artifacts');
+        return prepareArtifactsDir(artifactsDir)
+          .then(makeSureItFails())
+          .catch(onlyInstancesOf(UsageError, (error) => {
+            assert.match(error.message, /lack permissions/);
+          }));
+      });
+    }
+  ));
+
+  it('throws error when creating a folder if lacks writing permissions',
+    () => withTempDir(
+    (tmpDir) => {
+      class ErrorWithCode extends Error {
```

Comment:
If we decide to use this approach, it would be nice to move this definition of `ErrorWithCode` in a file shared with the other test file that is already using a similar class.